xformers/components/attention/csrc/cuda/mem_eff_attention/attention_backward_generic.cu

```diff
@@ -105,29 +116,60 @@ mem_efficient_attention_backward_cutlass(
     // TODO: Fuse this into a kernel?
     // This is a bottleneck for smaller sequences (M <= 128)
     auto delta = Kernel::kKernelComputesDelta
-        ? at::empty({B, M}, query.options().dtype(at::ScalarType::Float))
-        : (grad_out.to(at::kFloat) * out.to(at::kFloat)).sum(-1);
+        ? at::empty(
+              {B, num_heads, M}, query.options().dtype(at::ScalarType::Float))
+        : (grad_out.to(at::kFloat) * out.to(at::kFloat))
+              .sum(-1)
+              .transpose(-2, -1)
+              .contiguous();
```

Comment:
Now that you have this `contiguous()` call in here, maybe it might be the case that for floats it might be preferable to also dispatch to `kKernelComputesDelta` ?